packages/web-console/src/index.tsx

```diff
@@ -68,17 +69,19 @@ ReactDOM.render(
     <ScreenSizeProvider>
       <Provider store={store}>
         <SettingsProvider>
-          <AuthProvider>
-            <QuestProvider>
-              <GlobalStyle />
-              {ReactDOM.createPortal(<ToastContainer />, document.body)}
-              <LocalStorageProvider>
-                <FadeSlow />
-                <FadeReg />
-                <Layout />
-              </LocalStorageProvider>
-            </QuestProvider>
-          </AuthProvider>
+          <PosthogProviderWrapper>
```

Comment:
I'm starting to think we should wrap the app with `PostHogProvider` only when it's enabled in settings, so the code does not parse the provider if `posthog.api.key` is `null`. Not sure if this impacts app start time, but it's a hunch I have

Reply:
https://github.com/PostHog/posthog-js/blob/main/react/src/context/PostHogProvider.tsx 

The client is only initialised if the api key is provided. If the app is only conditionally wrapped, there will need to be additional checks if the client is undefined when the context is used. There are already [checks](https://github.com/PostHog/posthog-js/blob/108d8f88cf39926aa8d595fb39d14ddb7e4e8085/src/posthog-core.ts#L725-L727) in place within the client itself to ensure that the functions are no-ops if the client isn't initialised.

Reply:
OK, got it. Makes sense, I see they check for it